tgui/packages/tgui/interfaces/common/sorting/SortableTable/index.tsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author mloccy (https://github.com/mloccy)
+ * @license ISC
+ */
+
+import { BooleanLike } from "common/react";
+import { Fragment } from "inferno";
+import { useLocalState, useSharedState } from "../../../../backend";
+import { Table, Tooltip } from "../../../../components";
+import { Header } from "../Header";
+import { SearchState, SortableTableHeaderConfig, SortableTableRowProps, SortState } from "./type";
+import { onSortClick, sortAndFilterRows } from "./utils";
+
+
+interface SortableTableHeaderRowProps {
+  config: SortableTableHeaderConfig[],
+  sortState: SortState,
+  setSortBy: (index: number) => void
+}
+
+const SortableTableHeaderRow = (props: SortableTableHeaderRowProps) => {
+  const { config, sortState, setSortBy } = props;
+
+  return (
+    <Table.Row header>
+      {config.map((config, index: number) => {
+        const { children, sortable, toolTipContent } = config;
+        const needsHeader = sortable === undefined ? true : sortable;
+
+        const sortDirection = !sortState
+          ? null
+          : index === sortState.index
+            ? sortState.dir
+            : null;
+        const sortByIndex = () => setSortBy(index);
+
+
+        if (toolTipContent) {
+          if (needsHeader) {
+
+            return (
+              <Tooltip content={toolTipContent}>
+                <Table.Cell header key={index}>
+                  <Header sortDirection={sortDirection} onSortClick={sortByIndex}>{children}</Header>
+                </Table.Cell>
+              </Tooltip>);
+          }
+          else {
+            return (
+              <Tooltip content={toolTipContent}>
+                <Table.Cell header key={index}>
+                  {children}
+                </Table.Cell>
+              </Tooltip>);
+          }
+        } else {
+
+          if (needsHeader) {
+            return (
+              <Table.Cell header key={index}>
+                <Header sortDirection={sortDirection} onSortClick={sortByIndex}>{children}</Header>
+              </Table.Cell>);
+          } else {
+            return (
+              <Table.Cell header key={index}>
+                {children}
+              </Table.Cell>);
+
+          }
+        }
```

Comment:
This chunk here is pretty verbose and contains repeated parts. Consider separating out into wrapper and contents, with each being conditionally set to one of the two things, then combining together at the end.